### PR TITLE
conftest: fix version command

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

Fix `conftest --version` so it outputs the version

###### Things done

Now outputs the following:

```bash
$ conftest --version
Version: 0.21.0
Commit:
Date:
```

Also added a longDescription

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS (x86_64)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by yurrriq — APPROVED** [review]
lgtm, thanks!
```shell
λΠ nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/tarball/pull/99557/head -p conftest --run 'conftest --version'
unpacking 'https://github.com/NixOS/nixpkgs/tarball/pull/99557/head'...
these derivations will be built:
  /nix/store/n7qzhx2yv1r81578p0q318whaajnscbf-remove-references-to.drv
  /nix/store/c7qjsn8ibg62bwcngxxk3ynbxqkd3fla-conftest-0.21.0.drv
these paths will be fetched (94.48 MiB download, 418.09 MiB unpacked):
  /nix/store/18ggindri5xm8zbdpzy7x2927z1lg5gc-stdenv-linux
  /nix/store/3qhafmf9yj5nyyrnm9xnil6vl6gglsf5-bash-interactive-4.4-p23-info
  /nix/store/546gk8pnvdya2m2a6y791ppr71wj2imq-bash-interactive-4.4-p23-doc
  /nix/store/fabkq7vbl5x6q85rg150sxr1g3rixyys-go-1.15.2
  /nix/store/g5asxygf03fbkbr9h8sw15ac978bjppg-conftest-0.21.0-go-modules
  /nix/store/l2g6my7nykfshahijqnk7gy7s5lq2r3k-bash-interactive-4.4-p23-man
  /nix/store/ly74nfilzwx6raicvc53jdplhh4f6b0r-source
...
Version: 0.21.0
Commit: 
Date: 
```

**Comment by 06kellyjac** [comment]
/marvin opt-in
@yurrriq thanks for the review, could you post a comment with `/status needs_merger`? The bot needs someone who isn't the PR author to do it

**Comment by 06kellyjac** [comment]
Result of `nixpkgs-review pr 99557` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>conftest</li>
  </ul>
</details>

File: pkgs/development/tools/conftest/default.nix [hunk 1]
```diff
@@ -15,13 +15,25 @@ buildGoModule rec {
...
 
-  buildFlagsArray = ''
-    -ldflags=
-        -X main.version=${version}
-  '';
+  buildFlagsArray = [
+    "-ldflags="
+    "-s"
+    "-w"
+    "-X github.com/open-policy-agent/conftest/internal/commands.version=${version}"
```
Comment (kevincox) [inline pkgs/development/tools/conftest/default.nix:22]: Is it possible to document these? Using a long form would be ideal but if not just giving a couple of words each helps readers looking for something quickly.

**Comment by kevincox** [comment]
I'm going to megre for now. But feel free to send me a follow up with some comments if you think that is a good idea.

File: pkgs/development/tools/conftest/default.nix (hunk 1 above)
Comment (06kellyjac) [reply pkgs/development/tools/conftest/default.nix:22]: I'm not sure it's worth documenting `buildFlagsArray`/`ldflags` as most people packaging know what they are and they're easy to look up in `ld` documentation. As the variable name says they're flags for `ld`.
Also there are many many many packages with almost identical inputs that aren't documented.

I wouldn't add a comment above `vendorSha256` explaining what it's for or add a comment for why there's `rec` in `buildGoModule rec {`

I'm happy to remove `-s` and `-w` though as now that I think about it they're not particularly necessary, it was mostly leftover from another derivation

Comment (kevincox) [reply pkgs/development/tools/conftest/default.nix:22]: Yeah. Removing `-s` and `-w` makes sense, and removes two of the things to question. I honestly can't find out what `-X` is. I agree, `-ldflags` is obvious enough.

**Pull request merged**